beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1667,6 +1708,124 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         );
         Ok(())
     }
+
+    /// Try to prune blobs older than the data availability boundary.
+    pub fn try_prune_blobs(
+        &self,
+        force: bool,
+        data_availability_boundary: Option<Epoch>,
+    ) -> Result<(), Error> {
+        let blob_info = match self.get_blob_info() {
+            Some(blob_info) => blob_info,
+            None => {
+                return Ok(());
+            }
+        };
+
+        let oldest_blob_slot = blob_info.oldest_blob_slot;
+        // The last entirely pruned epoch, blobs sidecar pruning may have stopped early in the
+        // middle of an epoch.
+        let last_pruned_epoch = oldest_blob_slot.epoch(E::slots_per_epoch()) - 1;
+        let next_epoch_to_prune = match data_availability_boundary {
+            Some(epoch) => epoch,
+            None => {
```

Comment:
When `chain.data_availability_boundary()` outputs `None`, it indicates the 4844 fork isn't set, so there are no blobs and no need to prune. So I think this should be a no-op.  I can see this logic is useful when pruning via the database manager so you don't have to keep track of an epoch to manually prune. But we'll need to differentiate this use-case from the absence of the data availability boundary here.

An option is to require a `data_availability_boundary` from the cli. Then we could just make it a required param in this method and force handling the `None` case when calling this method.  maybe this is too dangerous though because we don't yet have a mechanism to re-sync only blobs if blocks vs blobs become inconsistent in the database. But I guess we will need that mechanism anyways to support [this](https://github.com/sigp/lighthouse/pull/3869). 

If we want to make the data availabilty boundary optional in the cli, this seems like reasonable fallback logic, I'd just point out it this can't tell us the current epoch, it can tell us the earliest possible current epoch if that makes sense. The chain usually finalizes in two epochs but when finalization doesn't advance this won't either  https://github.com/sigp/lighthouse/pull/3852/files#diff-5c52c8ceff6131bfe8a054b3f9a856eb9a264b4e6e95e4cd59e35bb70992f2e8R1732-R1736

Reply:
Why is a method to re-sync blobs connected to a blobs freezer? 

Paul reviewed that PR btw and the term 'blobs freezer' doesn't really apply anymore, it's just 'blobs db', as an optional different dir to store blobs in regardless of their slot relative to finality.

Reply:
> Why is a method to re-sync blobs connected to a blobs freezer?

If a user puts their blobs store is on it's own drive that gets corrupted, or accidentally deletes the blobs db, currently I think people would just have to fully re-sync blocks and blobs (as opposed to just blobs) which would be pretty bad UX. Generally making blobs it's own db seems to make it easier to have a discrepency between local blocks and blobs. Having a prune that would be relatively easy to mess up would similarly enable this. So being able to re-sync an arbitrary range of is a feature I think we'll need to add (not in this PR)